src/commands/link/link.ts

```diff
@@ -1,220 +1,205 @@
   const { api, state } = command.netlify
 
   const SITE_NAME_PROMPT = 'Search by full or partial site name'
   const SITE_LIST_PROMPT = 'Choose from a list of your recently updated sites'
   const SITE_ID_PROMPT = 'Enter a site ID'
-
   let GIT_REMOTE_PROMPT = 'Use the current git remote origin URL'
   let site
   // Get git remote data if exists
   const repoData = await getRepoData({ workingDir: command.workingDir, remoteName: options.gitRemoteName })
 
-  let linkChoices = [SITE_NAME_PROMPT, SITE_LIST_PROMPT, SITE_ID_PROMPT]
+  let linkOptions = [{ value: SITE_NAME_PROMPT }, { value: SITE_LIST_PROMPT }, { value: SITE_ID_PROMPT }]
 
   if (!repoData.error) {
     // Add git GIT_REMOTE_PROMPT if in a repo
     GIT_REMOTE_PROMPT = `Use current git remote origin (${repoData.httpsUrl})`
-    linkChoices = [GIT_REMOTE_PROMPT, ...linkChoices]
+    linkOptions = [{ value: GIT_REMOTE_PROMPT }, ...linkOptions]
   }
 
-  log()
-  log(`${chalk.cyanBright('netlify link')} will connect this folder to a site on Netlify`)
-  log()
-  const { linkType } = await inquirer.prompt([
-    {
-      type: 'list',
-      name: 'linkType',
-      message: 'How do you want to link this folder to a site?',
-      choices: linkChoices,
-    },
-  ])
+  NetlifyLog.message(`${chalk.cyanBright('netlify link')} will connect this folder to a site on Netlify`)
+
+  const linkType = await select({
+    message: 'How do you want to link this folder to a site?',
+    options: linkOptions,
+  })
 
   let kind
   switch (linkType) {
     case GIT_REMOTE_PROMPT: {
       kind = 'gitRemote'
-      log()
-      log(`Looking for sites connected to '${repoData.httpsUrl}'...`)
-      log()
+      const loading = spinner()
+      loading.start(`Looking for sites connected to '${repoData.httpsUrl}' (this can take a bit).`)
       const sites = await listSites({ api, options: { filter: 'all' } })
-
       if (sites.length === 0) {
-        error(`You don't have any sites yet. Run ${chalk.cyanBright('netlify sites:create')} to create a site.`)
+        loading.stop(
+          `You don't have any sites yet. Run ${chalk.cyanBright('netlify sites:create')} to create a site.`,
+          1,
+        )
+        outro({ exit: true })
       }
 
       const matchingSites = sites.filter(
-        // @ts-expect-error TS(2339) FIXME: Property 'repo_url' does not exist on type '{}'.
-        ({ build_settings: buildSettings = {} }) => repoData.httpsUrl === buildSettings.repo_url,
+        ({ build_settings: buildSettings = {} }: { build_settings: { repo_url?: string } }) =>
+          repoData.httpsUrl === buildSettings.repo_url,
       )
 
       // If no remote matches. Throw error
       if (matchingSites.length === 0) {
-        log(chalk.redBright.bold.underline(`No Matching Site Found`))
-        log()
-        log(`No site found with the remote ${repoData.httpsUrl}.
+        NetlifyLog.warn(chalk.bold(`No Matching Site Found`))
```

Comment:
I know this existed before but the casing on this is weird compared to what we do elsewhere.

```suggestion
        NetlifyLog.warn(chalk.bold(`No matching site found`))
```